azopsreference/3fc1081d-6105-4e19-b60c-1ec1252cf560/contoso/.AzState/Microsoft.Authorization_policyDefinitions-Deploy-Nsg-FlowLogs.parameters.json

```diff
@@ -23,6 +23,41 @@
               "metadata": {
                 "displayName": "Storage Account Resource Id"
               }
+            },
+            "trafficAnalyticsInterval": {
+              "type": "Integer",
+              "metadata": {
+                "displayName": "Traffic Analytics processing interval mins (10/60)"
+              },
+              "defaultValue": 60
+            },
+            "flowAnalyticsEnabled": {
+              "type": "Boolean",
+              "metadata": {
+                "displayName": "Enable Traffic Analytics"
+              },
+              "defaultValue": false
+            },
+            "flowAnalyticsWorkspaceId": {
+              "type": "string",
+              "metadata": {
+                "displayName": "Workspace GUID for Log Analytics"
```

Comment:
can you change this parameter to be "logAnalytics" and use "strongType": "oms" ?
That will give customer a better UX when assigning via portal

Reply:
Sure thing - I will update the flowAnalyticsWorkspaceResourceId parameter as it is this that refers to the ARM resource.

The parameter you have highlighted is the GUID of the workspace and is a separate property.

Reply:
Got it. If it's the workspaceId (not workspaceResourceId) then you can ignore the "strongType": "oms" for the parameter, and consider using the reference function instead (wrapped into if condition). So when providing the logAnalytics resourceId, ARM will retrieve the guid during runtime

Reply:
I also need this parameter to be optional - does strongType allow for empty entries?

Reply:
Will investigate the reference function too - thanks :)

Reply:
Am testing now - will report back later today

Reply:
This works well - thanks for the tips